APIRouter tags

euri10 opened the pull request:
Humble attempt at solving #54

euri10 committed:
Attempt at APIRouter tags

euri10 committed:
Tags at route level are appended to tags at APIRouter level

euri10 committed:
Avoided mypy warning

euri10 committed:
Black magic

euri10 committed:
Added more tests for coverage
Chmod +x lint

euri10 committed:
Tests parametrization for (hopefully) better coverage

euri10 committed:
Hopefully better coverage adding tags generation
on other methods

euri10 committed:
Head tested

tiangolo commented:
Awesome! That's a lot of work you put in there! :clap: :taco: 

---

I was thinking that we could put this logic in the `.include_router()` method.

We could change this line: https://github.com/tiangolo/fastapi/blob/master/fastapi/routing.py#L240

And add a parameter `tags: List[str] = None`.

And then change this line: https://github.com/tiangolo/fastapi/blob/master/fastapi/routing.py#L253

To something like:

`tags=(route.tags or []) + (tags or []),`

---

That way we don't have to add extra code to the path operation decorators and especially, we don't have to override the `APIRouter.__init__()` method. By not overriding it, it's a lot easier to support new versions of Starlette that can change it.

euri10 commented:
that makes a lot of sense, I'll modify that tomorrow

On Mon, Mar 4, 2019 at 7:40 PM Sebastián Ramírez <notifications@github.com>
wrote:

> Awesome! That's a lot of work you put in there! 👏 🌮
> ------------------------------
>
> I was thinking that we could put this logic in the .include_router()
> method.
>
> We could change this line:
> https://github.com/tiangolo/fastapi/blob/master/fastapi/routing.py#L240
>
> And add a parameter tags: List[str] = None.
>
> And then change this line:
> https://github.com/tiangolo/fastapi/blob/master/fastapi/routing.py#L253
>
> To something like:
>
> tags=(route.tags or []) + (tags or []),
> ------------------------------
>
> That way we don't have to add extra code to the path operation decorators
> and especially, we don't have to override the APIRouter.__init__()
> method. By not overriding it, it's a lot easier to support new versions of
> Starlette that can change it.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tiangolo/fastapi/pull/55#issuecomment-469368925>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABDZPuftAS7Jjlmjtvlc1IOiOUFkx_Nwks5vTWifgaJpZM4bPYEt>
> .
>


-- 
benoit barthelet
http://pgp.mit.edu/pks/lookup?op=get&search=0xF150E01A72F6D2EE

euri10 committed:
Modified tags to be at include_router level
Changed tests expectations accordingly as simple route tags are now **before** the router tags

euri10 committed:
Forgot blackmagic

tiangolo committed:
:memo: Update docs to include new APIRouter tags

tiangolo committed:
:fire: Remove test tags, that are now tested as part of docs

tiangolo committed:
:art: Format routing

tiangolo committed:
:rewind: Revert changing Pipfile.lock

tiangolo commented:
Great! I just added docs https://fastapi.tiangolo.com/tutorial/bigger-applications/ and a couple updates.

Great idea and good job! :tada: :taco: :cake: Thanks :clap:
